src/coreclr/gc/gc.cpp

```diff
@@ -6915,21 +6935,41 @@ bool gc_heap::virtual_commit (void* address, size_t size, int bucket, int h_numb
         check_commit_cs.Enter();
         bool exceeded_p = false;
 
-        if (heap_hard_limit_oh[bucket] != 0)
+        if ((heap_hard_limit_oh[soh] != 0) && (bucket < total_oh_count) && (heap_hard_limit_oh[bucket] != 0))
```

Comment:
do we really need to check `heap_hard_limit_oh[soh]`, then check `heap_hard_limit_oh[bucket]` again?